solidity/test/Coordinator_test.js

```diff
@@ -31,38 +47,72 @@ contract('Coordinator', () => {
 
   describe('#initiateServiceAgreement', () => {
     it('saves a service agreement struct from the parameters', async () => {
-      await coordinator.initiateServiceAgreement(
-        1,
-        2,
-        ['0x70AEc4B9CFFA7b55C0711b82DD719049d615E21d', '0xd26114cd6EE289AccF82350c8d8487fedB8A0C07'],
-        '0x85820c5ec619a1f517ee6cfeff545ec0ca1a90206e1a38c47f016d4137e801dd'
-      )
+      const payment = newHash('1000000000000000000')
+      const expiration = newHash('300')
+      const account = newAddress(web3.eth.accounts[0])
+      const oracles = [account]
+      const requestDigest = newHash('0x9ebed6ae16d275059bf4de0e01482b0eca7ffc0ffcc1918db61e17ac0f7dedc8')
+
+      const serviceAgreementIDInput = concatTypedArrays(
+        payment,
+        expiration,
+        // XXX: the account is an address, which is 20 bytes, but solidity
+        // abi.encodePacked treats it as a 32 byte hash :(
+        // concatTypedArrays(...oracles),
+        concatTypedArrays(...(oracles.map(a => newHash(toHex(a))))),
+        requestDigest)
+
+      const serviceAgreementIDInputDigest = utils.sha3(toHex(serviceAgreementIDInput))
+      const serviceAgreementID = newHash(toHex(serviceAgreementIDInputDigest))
 
-      let sa = await coordinator.serviceAgreements.call(
-        '0x2249a9e0a0463724551b2980299a16406da6f4e80d911628ee77445be4e04e7c'
+      let oracleSignature =
+        newSignature(web3.eth.sign(toHexWithoutPrefix(account), toHexWithoutPrefix(serviceAgreementID)))
+
+      const {v, r, s} = splitRPCSignature(oracleSignature)
```

Comment:
How do you feel about calling this `splitSignature`. `ethereumjs-util` used Rpc in the name but it doesn't matter where the signature came from.

Reply:
Hmm, I'm okay with it, but I think the RPC signatures have a trailing v, vs a leading v in other places.

Reply:
Ah, ok. I didn't realize there was an ordering difference. If that's the case seems fine to leave it.